pkg/virt-handler/mount-manager/mount.go

```diff
@@ -0,0 +1,109 @@
+package mount
+
+import (
+	"fmt"
+	"time"
+
+	"k8s.io/apimachinery/pkg/types"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	containerdisk "kubevirt.io/kubevirt/pkg/container-disk"
+	virtconfig "kubevirt.io/kubevirt/pkg/virt-config"
+	container_disk "kubevirt.io/kubevirt/pkg/virt-handler/container-disk"
+	hotplug_volume "kubevirt.io/kubevirt/pkg/virt-handler/hotplug-disk"
+	"kubevirt.io/kubevirt/pkg/virt-handler/isolation"
+)
+
+// MountInfo wraps all the mount information
+type MountInfo struct {
+	containerDisksInfo map[string]*containerdisk.DiskInfo
+}
+
+func (info *MountInfo) GetContainerDisksInfo() map[string]*containerdisk.DiskInfo {
+	return info.containerDisksInfo
+}
+
+// MountManager handles all the mount operations required by KubeVirt
+type MountManager interface {
+	Mount(vmi *v1.VirtualMachineInstance) (MountInfo, error)
+	Unmount(vmi *v1.VirtualMachineInstance) error
+	ContainerDiskMountsReady(vmi *v1.VirtualMachineInstance, notInitializedSince time.Time) (bool, error)
+	SyncHotplugMounts(vmi *v1.VirtualMachineInstance) error
```

Comment:
`SyncHotplugMounts` is called at different times than `Mounts` . Are you proposing to add a flag to distinguish the 2 cases? It is possible but for me, it just complicates the `Mounts` function. In the case, we will add a further mount type that can be added once the VM has started to me it would make sense to have a sync function.